### PR TITLE
Saving changes for groups page

### LINKED ISSUE #124: Create Backend API & Link to frontend for Rendering a Group Page
### Description

This User Story serves to setup the backend API requests in order to retrieve the necessary data from the database to render a single Group Page

### PR DESCRIPTION
**Pull request by JayC-SF**
Groups page now working

### CODE REVIEW

File: backend/.gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 tmp/
-.env
\ No newline at end of file
+.env
+backend
```
Comment (LamdaDev) [inline backend/.gitignore:3]: Just double checking, is backend being added into the .gitignore?

File: backend/models/group.go [hunk 2]
```diff
@@ -1,9 +1,144 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return &group, nil
+}
+
+func (g *Group) GetStudents() ([]User, error) {
+	collection := database.GetInstance().Database("RateMyPeersDB").Collection("UsersGroups")
+
+	pipeline := mongo.Pipeline{
```
Comment (LamdaDev) [inline backend/models/group.go:37]: Can you document this pipeline and write what would be the equivalent in SQL?

File: backend/models/group.go [hunk 3]
```diff
@@ -1,9 +1,144 @@
...
+	}
+	return users, nil
+}
+
+// gets the list of students in the same class as the group, but do not have a group.
+func (g *Group) GetPotentialStudents() ([]User, error) {
+
+	collection := database.GetInstance().Database("RateMyPeersDB").Collection("UsersCourses")
+
+	pipeline := mongo.Pipeline{
```
Comment (LamdaDev) [inline backend/models/group.go:84]: Same thing here, just so we don't forget how it works later on

File: frontend/src/pages/home/GroupPage.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    PostNewStudentInGroup({
+      groupid: groupid || "",
+      email: selectRef.current?.value || "",
+    });
+    refreshStudentList();
+  };
+
+  const refreshStudentList = () => {
+    GetGroupInfo(groupid || "").then(setGroupInfo);
+    GetStudendsWithoutGroup(groupid || "").then(setNewStudents);
```
Comment (LamdaDev) [inline frontend/src/pages/home/GroupPage.tsx]: typo in GetStudendsWithoutGroup, can you fix this?

**Review by LamdaDev — APPROVED** [review]
LGTM! Great job for this, left a couple of comments

File: backend/.gitignore (hunk 1 above)
Comment (JayC-SF) [reply backend/.gitignore:3]: that's the backend binary file when you run go build it's not any go file no worries :)

**Review by LamdaDev — APPROVED** [review]
LGTM! Great job JC you did really well

**Pull request merged**